egui_demo_lib/src/easy_mark/easy_mark_editor.rs

```diff
@@ -117,59 +117,27 @@ impl EasyMarkEditor {
 
 fn shortcuts(ui: &Ui, code: &mut dyn TextBuffer, ccursor_range: &mut CCursorRange) -> bool {
     let mut any_change = false;
-    for event in &ui.input().events {
-        if let Event::Key {
-            key,
-            pressed: true,
-            modifiers,
-        } = event
+    for (key, surrounding) in [
+        (Key::B, "*"), // *bold*
+        (Key::C, "`"), // `code`
+        (Key::I, "/"), // /italics/
+        (Key::L, "$"), // $subscript$
+        (Key::R, "^"), // ^superscript^
+        (Key::S, "~"), // ~strikethrough~
+        (Key::U, "_"), // _underline_
+    ] {
+        if ui
+            .input_mut()
+            .consume_key(egui::Modifiers::new().command(true), key)
         {
-            if modifiers.command_only() {
-                match &key {
-                    // toggle *bold*
-                    Key::B => {
-                        toggle_surrounding(code, ccursor_range, "*");
-                        any_change = true;
-                    }
-                    // toggle `code`
-                    Key::C => {
-                        toggle_surrounding(code, ccursor_range, "`");
-                        any_change = true;
-                    }
-                    // toggle /italics/
-                    Key::I => {
-                        toggle_surrounding(code, ccursor_range, "/");
-                        any_change = true;
-                    }
-                    // toggle $lowered$
-                    Key::L => {
-                        toggle_surrounding(code, ccursor_range, "$");
-                        any_change = true;
-                    }
-                    // toggle ^raised^
-                    Key::R => {
-                        toggle_surrounding(code, ccursor_range, "^");
-                        any_change = true;
-                    }
-                    // toggle ~strikethrough~
-                    Key::S => {
-                        toggle_surrounding(code, ccursor_range, "~");
-                        any_change = true;
-                    }
-                    // toggle _underline_
-                    Key::U => {
-                        toggle_surrounding(code, ccursor_range, "_");
-                        any_change = true;
-                    }
-                    _ => {}
-                }
-            }
-        }
+            toggle_surrounding(code, ccursor_range, surrounding);
```

Comment:
According to the docs of `consume_key`, it will return true both when the key is pressed and when it is released, so this should toggle twice, which is not intentional. This is why I think `consume_key` should only detect presses and ignore releases (or the other way around)

Reply:
It only returns true when the key was pressed but it will remove the release event too.

Reply:
I haven't run in since last review :)

But then the docs for `consume_key` should be changed to indicate it is only triggered on press, and also match for `pressed: true`.